### ISSUE TITLE
Config file seems to be ignored when using "cfssl selfsign"

### ISSUE THREAD

**Issue by binaryvisions** [cloudflare/cfssl#889]
I am trying to add a CRL to a root CA, which issue #41 suggests can now be accomplished using the "selfsign" command.

However, it seems to ignore the entire contents of my config file. If my config file is malformed, it will throw an error, but it is disregarding the entire contents.

For a given **rootca_csr.json**
`{
  "CN": "My Root CA",
  "key": {
    "algo": "ecdsa",
	"size": 384
  },
  "names": [
    {
      "C": "US",
      "O": "My Company"
    }
  ],
  "ca": {
    "expiry": "262800h"
  }
}`

And a given **rootca_config.json**

`{
 "signing": {
   "default": {
     "crl_url": "http://my.crl.website/root.crl",
     "usages": [
       "signing",
       "key encipherment",
       "cert sign",
       "crl sign"
      ],
      "expiry": "262800h",
      "ca_constraint": {
        "is_ca": true,
        "max_path_len": 0,
        "max_path_len_zero": true
      }
    }
  }
}
`

And the given command:
`cfssl selfsign -config="rootca_config.json" -loglevel=0 "My Root CA" rootca_csr.json | cfssljson -bare rootca`

I get this:
`2018/05/01 19:51:02 [DEBUG] loading configuration file from rootca_config.json
2018/05/01 19:51:02 [DEBUG] parse expiry in profile
2018/05/01 19:51:02 [DEBUG] expiry is valid
2018/05/01 19:51:02 [DEBUG] validating configuration
2018/05/01 19:51:02 [DEBUG] validate local profile
2018/05/01 19:51:02 [DEBUG] profile is valid
2018/05/01 19:51:02 [DEBUG] configuration ok
2018/05/01 19:51:02 [INFO] generate received request
2018/05/01 19:51:02 [INFO] received CSR
2018/05/01 19:51:02 [INFO] generating key: ecdsa-384
2018/05/01 19:51:02 [DEBUG] generate key from request: algo=ecdsa, size=384
2018/05/01 19:51:02 [INFO] encoded CSR
*** WARNING ***

Self-signed certificates are dangerous. Use this self-signed
certificate at your own risk.

It is strongly recommended that these certificates NOT be used
in production.

*** WARNING ***
`

The resulting certificate, though, has a 4 month expiration, none of the correct usages, and no CRL. If I, say, rename the default section of that config file and run the command, I'll get errors about no default section, so it's definitely seeing the config file. Why isn't it processing the parameters?

**Comment by kisom** [comment]
Agreed, I think even better would be a 're-sign' command to adjust CA certificates.

**Comment by eedwards-sk** [comment]
What's the status on this?

This problem affects more than just `cfssl selfsign`, it also affects `cfssl gencert -initca`

given `snapkitchen.json`

```
{
  "key": {
    "algo": "ecdsa",
    "size": 256
  },
  "names": [
    {
      "C": "US",
      "L": "Austin",
      "O": "Snap Kitchen, LLC.",
      "OU": "DevOps",
      "ST": "Texas"
    }
  ]
}
```

given `config.json`


```
{
  "signing": {
    "default": {
        "expiry": "2160h"
    },
    "profiles": {
      "root": {
        "expiry": "87600h",
        "usages": [
          "cert sign",
          "crl sign"
        ],
        "ca_constraint": {
          "is_ca": true,
          "max_path_len": 1
        }
      }
    }
  }
}
```

generating the root-ca keypair:

input

```
→ cfssl gencert -initca=true -config=config.json -profile=root -cn="Root CA" -loglevel=0 snapkitchen.json | cfssljson -bare root-ca
```

output

```
2018/09/06 19:15:48 [DEBUG] loading configuration file from config.json
2018/09/06 19:15:48 [DEBUG] parse expiry in profile
2018/09/06 19:15:48 [DEBUG] expiry is valid
2018/09/06 19:15:48 [DEBUG] parse expiry in profile
2018/09/06 19:15:48 [DEBUG] expiry is valid
2018/09/06 19:15:48 [DEBUG] parse expiry in profile
2018/09/06 19:15:48 [DEBUG] expiry is valid
2018/09/06 19:15:48 [DEBUG] validating configuration
2018/09/06 19:15:48 [DEBUG] validate local profile
2018/09/06 19:15:48 [DEBUG] profile is valid
2018/09/06 19:15:48 [DEBUG] validate local profile
2018/09/06 19:15:48 [DEBUG] profile is valid
2018/09/06 19:15:48 [DEBUG] validate local profile
2018/09/06 19:15:48 [DEBUG] profile is valid
2018/09/06 19:15:48 [DEBUG] configuration ok
2018/09/06 19:15:48 [INFO] generating a new CA key and certificate from CSR
2018/09/06 19:15:48 [INFO] generate received request
2018/09/06 19:15:48 [INFO] received CSR
2018/09/06 19:15:48 [INFO] generating key: ecdsa-256
2018/09/06 19:15:48 [DEBUG] generate key from request: algo=ecdsa, size=256
2018/09/06 19:15:48 [INFO] encoded CSR
2018/09/06 19:15:48 [DEBUG] validating configuration
2018/09/06 19:15:48 [DEBUG] validate local profile
2018/09/06 19:15:48 [DEBUG] profile is valid
2018/09/06 19:15:48 [INFO] signed certificate with serial number 12795006206165776656655161890273867031595074482
```

confirming the expiration date is only 5 years (using the default value) rather than the specified value in the `root` profile:

```
→ cfssl certinfo -cert root-ca.pem | jq -r .not_before,.not_after
2018-09-07T00:11:00Z
2023-09-06T00:11:00Z
```

(same issue if I don't use a profile and just put everything in the `default` section, like the OP)
